crypto_hasher.go

```diff
@@ -110,9 +110,7 @@ func (ch *cryptoHasher) Write(b []byte) (n int, err error) {
 }
 
 func (ch *cryptoHasher) Sum(b []byte) []byte {
-	return ch.hashAlgorithm.Sum(b)
-}
-
-func (ch *cryptoHasher) Reset() {
```

Comment:
This seems to remove Reset() from the API. Is this intentional?

Reply:
I removed it because the crypto hasher APIs in other languages didn't have it (sum() would of reset the hash). 

However after looking at the golang hash API, it states sum() will not change the underlying state: https://golang.org/pkg/hash/

So to be idiomatic to Go, I added Reset() to IonHasher interface